compiler/il/symbol/OMRResolvedMethodSymbol.cpp

```diff
@@ -579,11 +582,54 @@ OMR::ResolvedMethodSymbol::genInduceOSRCallNode(TR::TreeTop* insertionPoint,
    return insertionPoint->getPrevTreeTop();
    }
 
+/*
+ *  \brief Generate the helper call to request for OSR transition and recompilation of the current compiled method.
+ *
+ *  \return true if the OSR transition is supported at the requested \parm induceBCI and the OSR induction helper call
+ *          is generated successfully
+ */
+bool
+OMR::ResolvedMethodSymbol::induceOSRAfterAndRecompile(TR::TreeTop *insertionPoint, TR_ByteCodeInfo induceBCI, TR::TreeTop* branch,
+    bool extendRemainder, int32_t offset, TR::TreeTop ** lastTreeTop)
+   {
+   TR_ASSERT(self()->comp()->allowRecompilation(), "request OSR and recompilation at node %p when recomp is disabled\n", insertionPoint);
+   TR::TreeTop *induceOSRCallTree = induceOSRAfterImpl(insertionPoint, induceBCI, branch, extendRemainder, offset, lastTreeTop);
+   if (!induceOSRCallTree)
+      return false;
+   TR::Node *induceOSRCallNode = induceOSRCallTree->getNode()->getFirstChild();
+   TR::SymbolReference *symRef = induceOSRCallNode->getSymbolReference();
+   TR_ASSERT_FATAL(induceOSRCallNode->getOpCode().isCall() &&
+                   symRef->getReferenceNumber() == TR_induceOSRAtCurrentPC,
+                   "induceOSRCallNode %p (n%dn) under induceOSRCallTree %p should be a call node with TR_induceOSRAtCurrentPC helper call", induceOSRCallNode, induceOSRCallNode->getGlobalIndex(), induceOSRCallTree->getNode());
+   induceOSRCallNode->setSymbolReference(self()->comp()->getSymRefTab()->findOrCreateRuntimeHelper(TR_induceOSRAtCurrentPCAndRecompile,
+                                                                                                   true /* canGCandReturn */,
+                                                                                                   true /* canGCandExcept */,
+                                                                                                   true /* preservesAllRegisters */));
+   return true;
+   }
 
+/*
+ *  \brief Generate the helper call to request for OSR transition only.
+ *
+ *  \return true if the OSR transition is supported at the requested \parm induceBCI and the OSR induction helper call
+ *          is generated successfully
+ */
 bool
 OMR::ResolvedMethodSymbol::induceOSRAfter(TR::TreeTop *insertionPoint, TR_ByteCodeInfo induceBCI, TR::TreeTop* branch,
     bool extendRemainder, int32_t offset, TR::TreeTop ** lastTreeTop)
    {
+   return induceOSRAfterImpl(insertionPoint, induceBCI, branch, extendRemainder, offset, lastTreeTop) != NULL;
+   }
+
+/*
+ *  \brief Internal implementation for generating the helper call to request for OSR transition.
+ *
+ *  \return The treetop of the inserted induceOSR helper call
```

Comment:
Sorry, one other knit: can this comment also specify under what conditions `NULL` is returned?

Reply:
added explanation for return value NULL. thanks!